core/src/physics.cpp

```diff
@@ -0,0 +1,73 @@
+#include <knoting/components.h>
+#include <knoting/engine.h>
+#include <knoting/physics.h>
+#include <knoting/scene.h>
+
+PxDefaultAllocator gAllocator;
+PxDefaultErrorCallback gErrorCallback;
+
+namespace knot {
+Physics::Physics(Engine& engine) : m_engine(engine) {
+    mPhysics = NULL;
+    mScene = NULL;
+    mFoundation = NULL;
+    mDispatcher = NULL;
```

Comment:
If you could initialize them just like `m_engine` (forgot the technical name of constructor definitions) and instead of `NULL` use C++ variant `nullptr`